apache/superset
# feat(RLS): RESTful apis and react view for RLS

### SUMMARY

Update RLS view to use React and REST apis.

this PR only migrates old view to new without any additional features like,
* ownership
* import export support.
* toggle like in alerts/reports to enable/disable RLS rules.
can add this in a separate PR.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


1. Empty state

![Screenshot from 2022-12-10 15-19-13](https://user-images.githubusercontent.com/12967587/206847208-618bbddb-b9de-4f27-b155-730951a246cd.png)

2. Add rule modal

![Screenshot from 2022-12-12 06-55-45](https://user-images.githubusercontent.com/12967587/206942283-d5ec0332-b2a2-4620-ba49-2fe61944be0c.png)



3. Populated list

![Screenshot from 2022-12-10 15-21-18](https://user-images.githubusercontent.com/12967587/206847416-33ebbde8-bd16-445b-8123-0a973984992f.png)

4. Bulk select to delete

![Screenshot from 2022-12-10 15-21-44](https://user-images.githubusercontent.com/12967587/206847496-6e06fafe-60cd-431c-9fc2-b68f70ec2c18.png)


### TESTING INSTRUCTIONS

1. Before merging this PR, create few RLS rules.
2. Then merge PR and check they are correctly shown on the new UI.
3. Delete all, create new and check correctly shown.
4. test filters in the list view.
6. Create chart on the table rule is set with the role and check its correctly getting applied.
7. Set `RLS_FILTER_RELATED_FIELDS` in the config.py as below
```
{
    "tables": [["table_name", filters.FilterStartsWith, "birth"]],
    "roles": [["name", filters.FilterContains, "Admin"]]"
}
```
8. When adding a new rule, in the table dropdown table names starting with `birth` should be shown and roles dropdown should only show roles containing text `Admin`.


### ADDITIONAL INFORMATION
